src/main/java/uk/gov/hmcts/reform/hmc/api/services/HearingsServiceImpl.java

```diff
@@ -490,4 +504,41 @@ public Hearings getFutureHearings(String caseReference) {
 
         return futureHearingsResponse;
     }
+
+    @Override
+    public HearingResponse createAutomatedHearings(CaseData caseData) {
+
+        final String userToken = idamTokenGenerator.generateIdamTokenForHearingCftData();
+        final String s2sToken = authTokenGenerator.generate();
+        AutomatedHearingRequest hearingRequest = AutomatedHearingTransformer.mappingHearingTransactionRequest(
+            caseData, ccdBaseUrl, specialCharacters);
+        printRequest(hearingRequest); // has to remove once all test completed
+        HearingResponse hearingResponse = hearingApiClient.createHearingDetails(
+            userToken,
+            s2sToken,
+            hearingRequest
+        );
+        return HearingResponse.builder()
+            .status(hearingResponse.getStatus())
+            .versionNumber(hearingResponse.getVersionNumber())
+            .hearingRequestID(hearingResponse.getHearingRequestID())
+            .timeStamp(hearingResponse.getTimeStamp())
+            .build();
+    }
+
+    private void printRequest(AutomatedHearingRequest hearingRequest) {
+
+        ObjectMapper objectMappers = new ObjectMapper();
+        objectMappers.registerModule(new JavaTimeModule());
+        try {
+            String automatedHearingRequestJson = objectMappers.writerWithDefaultPrettyPrinter().writeValueAsString(
+                hearingRequest);
+            log.info(
+                "Automated Hearing Request: createAutomatedHearings: hearingRequest: {}",
```

Comment:
will it have PI data?

Reply:
I am removing loggers with case data or any PI data, will remove this

Reply:
Done